src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,190 @@
+     * @return: Returns an integer that indicate the result of the run.
+     */
+    public int run(String inputString, Ui ui, TaskList taskList) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/M/yyyy");
+        String[] args = inputString.split("\\s+");
+        String action = args[0];
+
+        try {
+            validate(inputString, action, args, taskList);
+        } catch (DukeException de) {
+            System.out.println(de.getMessage());
+            return 1;
+        }
+
+        switch(action) {
+        case "bye":
+            ui.print("Bye. Hope to see you again soon!");
+            return -1;
+        case "list":
+            if (args.length > 1) {
+                switch (args[1]) {
+                case "/on":
+                    ui.print(taskList.getTasksBasedOnDate(LocalDate.parse(args[2],
+                            DateTimeFormatter.ofPattern("dd/M/yyyy")), 0));
+                    break;
+                case "/before":
+                    ui.print(taskList.getTasksBasedOnDate(LocalDate.parse(args[2],
+                            DateTimeFormatter.ofPattern("dd/M/yyyy")), 1));
+                    break;
+                case "/after":
+                    ui.print(taskList.getTasksBasedOnDate(LocalDate.parse(args[2],
+                            DateTimeFormatter.ofPattern("dd/M/yyyy")), 2));
+                    break;
+                default:
+                    break;
+                }
+            } else {
+                ui.print(taskList.getList());
+            }
+            return 0;
+        case "mark":
+            taskList.getTask(Integer.parseInt(args[1]) - 1).mark();
+            ui.print("Task marked as done: ", " " + taskList.getTask(Integer.parseInt(args[1]) - 1).toString());
+            return 1;
+        case "unmark":
+            taskList.getTask(Integer.parseInt(args[1]) - 1).unmark();
+            ui.print("Task as not done yet: ", " " + taskList.getTask(Integer.parseInt(args[1]) - 1).toString());
+            return 1;
+        case "todo":
+            taskList.addTask((inputString).substring(5).trim(), false, null, 0);
+            ui.print("Added Task: ", " " + taskList.getLast().toString(),
+                    String.format("There are now %d task(s) in the list.", taskList.getSize()));
+            return 1;
+        case "deadline":
+            taskList.addTask(inputString.substring(9).split("/by")[0].trim(), false,
+                    LocalDate.parse(inputString.split("/by")[1].substring(1), formatter), 1);
+            ui.print("Added Task: ", " " + taskList.getLast().toString(),
+                    String.format("There are now %d task(s) in the list.", taskList.getSize()));
+            return 1;
+        case "event":
+            taskList.addTask(inputString.substring(6).split("/at")[0].trim(), false,
+                    LocalDate.parse(inputString.split("/at")[1].substring(1), formatter), 2);
```

Comment:
Consider breaking into multiple lines with intermediate variables for readability.
```suggestion
            String taskDescription = inputString.substring(6).split("/at")[0].trim();
            LocalDate taskDate = LocalDate.parse(inputString.split("/at")[1].substring(1), formatter);
            taskList.addTask(taskDescription, false, taskDate, 2);
```